addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1808,4 +1808,42 @@ options.registry.CoverProperties = options.Class.extend({
         this.$target[0].dataset.filterColor = activeFilterColor || '';
     },
 });
+
+/**
+ * Whether the section should be full-width (container-fluid) or use a classic container
+ */
+options.registry.section_stretch = options.Class.extend({
+    /**
+     * Only show this option if it changes the visual
+     * @override
+     */
+    start: function () {
+        const originalTarget = this.$target;
+
+        const unstretchedTarget = this.$target
+            .clone()
+            .removeClass('container-fluid')
+            .addClass('container');
+        // Elements have to be appended in the DOM to have a clientWidth
+        this.$target.replaceWith(unstretchedTarget);
+        const unstretchedWidth = unstretchedTarget[0].clientWidth;
+        
+        const stretchedTarget = this.$target
+            .clone()
+            .removeClass('container')
+            .addClass('container-fluid');
+        unstretchedTarget.replaceWith(stretchedTarget);
+        const stretchedWidth = stretchedTarget[0].clientWidth;
+
+        const sizeDifference = stretchedWidth/unstretchedWidth - 1;
+        console.log('Size difference:', sizeDifference);
+        // The cutoff for the option is 5% difference in width
+        if(sizeDifference < 5/100) {
```

Comment:
You will have to install a linter to pass my crazy reviews :) https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines.

Here: `if (sizeDifference < 0.05) {` (missing space after if and around operators)